tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/AspNet/AspNetWebApi2Tests.cs

```diff
@@ -141,44 +149,47 @@ public AspNetWebApi2Tests(IisFixture iisFixture, ITestOutputHelper output, bool
 
         protected virtual string ExpectedServiceName => "sample";
 
-        public static TheoryData<string, int, int> Data() => new()
+        // classicMode doesn't change the paths we test, just the expectations, so can use either
+        public static IEnumerable<object[]> Data() => IntegratedExpectations.Keys.Select(x => new object[] { x });
+
+        public static Dictionary<string, (int StatusCode, int ExpectedSpanCount)> CreateExpectations(bool classicMode) => new()
         {
-            { "/api/environment", 200, 2 },
-            { "/api/absolute-route", 200, 2 },
-            { "/api/delay/0", 200, 2 },
-            { "/api/delay-optional", 200, 2 },
-            { "/api/delay-optional/1", 200, 2 },
-            { "/api/delay-async/0", 200, 2 },
-            { "/api/transient-failure/true", 200, 2 },
-            { "/api/transient-failure/false", 500, 3 },
-            { "/api/statuscode/201", 201, 2 },
-            { "/api/statuscode/503", 503, 2 },
-            { "/api/constraints", 200, 2 },
-            { "/api/constraints/201", 201, 2 },
-            { "/api/TransferRequest/401", 401, 4 },
-            { "/api/TransferRequest/503", 503, 4 },
-            { "/api2/delay/0", 200, 2 },
-            { "/api2/optional", 200, 2 },
-            { "/api2/optional/1", 200, 2 },
-            { "/api2/delayAsync/0", 200, 2 },
-            { "/api2/transientfailure/true", 200, 2 },
-            { "/api2/transientfailure/false", 500, 3 },
-            { "/api2/statuscode/201", 201, 2 },
-            { "/api2/statuscode/503", 503, 3 },
+            { "/api/environment", (200, 2) },
+            { "/api/absolute-route", (200, 2) },
+            { "/api/delay/0", (200, 2) },
+            { "/api/delay-optional", (200, 2) },
+            { "/api/delay-optional/1", (200, 2) },
+            { "/api/delay-async/0", (200, 2) },
+            { "/api/transient-failure/true", (200, 2) },
+            { "/api/transient-failure/false", (500, 3) },
+            { "/api/statuscode/201", (201, 2) },
+            { "/api/statuscode/503", (503, 2) },
+            { "/api/constraints", (200, 2) },
+            { "/api/constraints/201", (201, 2) },
+            { "/api/TransferRequest/401", (401, classicMode ? 2 : 4) },
```

Comment:
Aha I couldn't figure how why these were going from 2 to 4, but that explains it